### PR TITLE
Implement CertRenewEvent

### PR DESCRIPTION
**Pull request by elcore**
### 1. What does this change do, exactly?

1. Register a new Event Hook `CertRenewEvent`

### 2. Please link to the relevant issues.

#1864 

### 3. Which documentation changes (if any) need to be made because of this PR?

/

### 4. Checklist

- [ ] I have written tests and verified that they fail without my change
- [X] I have squashed any insignificant commits
- [X] This change has comments for package types, values, functions, and non-obvious lines of code
- [X] I am willing to help maintain this change if there are issues with it later

### CODE REVIEW

**Comment by elcore** [comment]
**NOTE: Merge this PR before #1864**

File: caddytls/client.go [hunk 1]
```diff
@@ -331,6 +331,9 @@ func (c *ACMEClient) Renew(name string) error {
 		return errors.New("too many renewal attempts; last error: " + err.Error())
 	}
 
+	// Executes Cert renew events
+	caddy.EmitEvent(caddy.CertRenewEvent, nil)
```
Comment (mholt) [inline caddytls/client.go]: What should we pass as data in the second argument?

File: caddy/caddymain/run.go [hunk 2]
```diff
@@ -126,6 +126,9 @@ func Run() {
 		mustLogFatalf("%v", err)
 	}
 
+	// Executes Instance Startup events
+	caddy.EmitEvent(caddy.InstanceStartupEvent, nil)
```
Comment (mholt) [inline caddy/caddymain/run.go]: What should we pass as data to this event?

Comment (elcore) [reply caddy/caddymain/run.go]: instance

File: caddytls/client.go (hunk 1 above)
Comment (elcore) [reply caddytls/client.go]: Nothing

Comment (mholt) [reply caddytls/client.go]: But won't hooks want to know which certificate was renewed, at least?

Comment (elcore) [reply caddytls/client.go]: Great idea!

**Pull request merged**